posthog/queries/util.py

```diff
@@ -171,3 +171,25 @@ def start_of_week_fix(filter: Filter) -> str:
     This function adds mode to the trunc_func, but only if the interval is week
     """
     return "0," if filter.interval == "week" else ""
+
+
+# TODO: Improve the typing here
+def filter_with_search_properties(filter: Any) -> Any:
+    search = getattr(filter, "search", None)
+    if not search:
+        return filter
+
+    new_group = {
+        "type": "OR",
+        "values": [
+            {"key": "email", "type": "person", "value": search, "operator": "icontains"},
+            {"key": "name", "type": "person", "value": search, "operator": "icontains"},
+            {"key": "distinct_id", "type": "event", "value": search, "operator": "icontains"},
+        ],
+    }
+    prop_group = (
+        {"type": "AND", "values": [new_group, filter.property_groups.to_dict()]}
+        if filter.property_groups.to_dict()
+        else new_group
+    )
+    return filter.with_data({"properties": prop_group})
```

Comment:
this should set `search: None` as well, because it's possible to have a `PersonQuery` being accessed from the `ActorBaseQuery`, which would imply a recursive search which might be inefficient/ not something we want/ possible to introduce subtle bugs